Src/Web/Web.Shared.Net/ExceptionTrackingTelemetryModule.cs

```diff
@@ -67,8 +75,47 @@ public void OnError(HttpContext context)
         /// <param name="configuration">Telemetry configuration to use for initialization.</param>
         public void Initialize(TelemetryConfiguration configuration)
         {
-            this.telemetryClient = new TelemetryClient(configuration);
-            this.telemetryClient.Context.GetInternalContext().SdkVersion = SdkVersionUtils.GetSdkVersion("web:");
+            if (this.EnableMvcAndWebApiExceptionAutoTracking)
+            {
+                if (!this.isInitialized)
+                {
+                    lock (this.lockObject)
+                    {
+                        if (!this.isInitialized)
+                        {
+                            this.telemetryClient = new TelemetryClient(configuration);
+                            this.telemetryClient.Context.GetInternalContext().SdkVersion = SdkVersionUtils.GetSdkVersion("web:");
+                            ExceptionHandlersInjector.Inject(this.telemetryClient);
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// IDisposable implementation.
+        /// </summary>
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// IDisposable implementation.
+        /// </summary>
+        /// <param name="disposing">The method has been called directly or indirectly by a user's code.</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!this.disposed)
+            {
+                if (disposing)
+                {
+                    this.isInitialized = false;
```

Comment:
Why reset this? Do you want to allow re-initialization after Dispose? (Perhaps Initialize should check `disposed` and throw an `ObjectDisposedException`

Reply:
it's for testing only - as we need o initialize module more than once in the same process.

Reply:
while I can test underlying ExceptionHandlersInjector, testing it does not guarantee that it's ever called. So, in any case, I would need Dispose/clear logic on the ExceptionTrackingTelemetryModule to verify it calls ExceptionHandlersInjector.

I can substitute public dispose with internal `isInitialized ` field.
I believe almost all modules are disposables and it fits into the model better than internal fields accessible by tests.

Thoughts?